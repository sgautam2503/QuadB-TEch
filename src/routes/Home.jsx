import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  Card, Container, Row, Spinner,
} from 'react-bootstrap';
import { fetchShowsFromAPI } from '../redux/shows/showsSlice';
import Shows from '../components/Shows';
import Searchbar from '../components/Searchbar';

function Home() {
  const [showName, setShowName] = useState('');

  const handleSearchShow = (e) => {
    setShowName(e.target.value.toLowerCase());
  };
  const showsItems = useSelector((state) => {
    const items = Object.entries(state)[0][1];
    return items;
  });
  const isLoading = useSelector((state) => {
    const items = Object.entries(state)[1][1];
    return items;
  });

  const filterShow = showsItems.filter((show) => show.name.toLowerCase().trim().includes(showName.trim()));

  const filteredShow = Object.entries(filterShow);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchShowsFromAPI());
  }, [dispatch]);

  return (
    <>
      <Container
        style={{
          minHeight: 'calc(100vh - 8vh)',
          width: '100vw',
        }}
        className="pb-3"
      >
        {isLoading && (
          <Card
            style={{
              height: 'calc(100vh - 8vh)',
              backgroundColor: '#4369b2',
              borderColor: '#4369b2',
            }}
            className="justify-content-center"
          >
            <Spinner
              animation="border"
              variant="light"
              className="mx-auto my-3"
            />
          </Card>
        )}
        <Container className="p-0">
          {!isLoading && (
            <Searchbar
              showName={showName}
              handleSearchShow={handleSearchShow}
            />
          )}
          <Row className="p-0">
            {!isLoading
            && (filteredShow.length === 0
              || filteredShow.length === showsItems.length)
              ? showsItems.map((show) => (
                <Shows
                  key={show.id}
                  id={show.id}
                  name={show.name}
                  rating={show.rating.average}
                  image={show.image.medium}
                  detailsStatus={show.detailsStatus}
                />
              ))
              : filteredShow.map((show) => (
                <Shows
                  key={show[1].id}
                  id={show[1].id}
                  name={show[1].name}
                  rating={show[1].rating.average}
                  image={show[1].image.medium}
                  detailsStatus={show[1].detailsStatus}
                />
              ))}
          </Row>
        </Container>
      </Container>
    </>
  );
}

export default Home;
