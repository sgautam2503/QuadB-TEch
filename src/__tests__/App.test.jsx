import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from '../routes/Home';
import store from '../redux/Store';

describe('App component', () => {
  it('should display on the screen', () => {
    const appComponent = render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    expect(appComponent).toMatchSnapshot();
  });
});
