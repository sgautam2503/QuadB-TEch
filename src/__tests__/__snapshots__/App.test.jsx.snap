// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > should display on the screen 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pb-3 card"
        style="background-color: rgb(13, 230, 253); min-height: calc(100vh - 8vh);"
      >
        <div
          class="justify-content-center card"
          style="height: calc(100vh - 8vh);"
        >
          <div
            class="mx-auto my-3 spinner-border text-dark"
          />
        </div>
        <div
          class="container"
        >
          <div
            class="row"
            style="background-color: rgb(13, 230, 253);"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pb-3 card"
      style="background-color: rgb(13, 230, 253); min-height: calc(100vh - 8vh);"
    >
      <div
        class="justify-content-center card"
        style="height: calc(100vh - 8vh);"
      >
        <div
          class="mx-auto my-3 spinner-border text-dark"
        />
      </div>
      <div
        class="container"
      >
        <div
          class="row"
          style="background-color: rgb(13, 230, 253);"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
